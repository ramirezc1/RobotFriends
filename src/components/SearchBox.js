import React from "react";

const SearchBox = ({ onSearchChange }) => {
  return (
    <div className="pa3">
      <input
        aria-label="Search"
        onChange={onSearchChange}
        className="pa3 ba b--green bg-lightest-blue"
        type="search"
        placeholder="search robots"
        name="searchField"
      />
    </div>
  );
};

export default SearchBox;
