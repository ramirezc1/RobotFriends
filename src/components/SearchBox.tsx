import React from "react";
interface ISearchBoxProps {
  onSearchChange(event: React.SyntheticEvent<HTMLInputElement>):void 
}

const SearchBox = ({ onSearchChange }:ISearchBoxProps) => {
  return (
    <div className="pa3">
      <input
        aria-label="Search"
        onChange={onSearchChange}
        className="pa3 ba b--green bg-lightest-blue"
        type="search"
        placeholder="search robots"
        name="searchField"
      />
    </div>
  );
};

export default SearchBox;
