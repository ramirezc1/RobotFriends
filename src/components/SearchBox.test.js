import { shallow } from "enzyme";
import SearchBox from "./SearchBox";


it("expect to render SearchBox component", () =>{
  const mockFn = {
    onSearchChange: jest.fn()
  };
  expect(shallow(<SearchBox {...mockFn}></SearchBox>)).toMatchSnapshot();
})

