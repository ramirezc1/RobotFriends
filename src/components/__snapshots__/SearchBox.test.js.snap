// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render SearchBox component 1`] = `
<div
  className="pa3"
>
  <input
    aria-label="Search"
    className="pa3 ba b--green bg-lightest-blue"
    name="searchField"
    onChange={[MockFunction]}
    placeholder="search robots"
    type="search"
  />
</div>
`;
